src/Core/BinCodec.fs

```diff
@@ -0,0 +1,72 @@
+﻿namespace B2R2
+
+/// Little Endian Base 128 Codec
+module LEB128 =
+  [<Literal>]
+  let Max32LEB128Length = 5
+  [<Literal>]
+  let Max64LEB128Length = 10
+
+  let DecodeUInt32 (bytes: byte []) =
+    let rec decodeLoop (offset, value, currentByte, length) =
+      let newValue = value ||| ((uint32 currentByte &&& 0x7fu) <<< (offset * 7))
+      if currentByte &&& 0x80uy = 0uy || offset = length - 1 then
+        Ok newValue
+      else
+        decodeLoop (offset + 1, newValue, bytes.[offset + 1], length)
```

Comment:
Use currying instead of using a tuple.

Also, can you move this loop out of the parent function?

Reply:
I'll try